lib/spack/spack/test/repo.py

```diff
@@ -90,14 +91,44 @@ def test_namespace_hasattr(attr_name, exists, mutable_mock_repo):
     assert hasattr(nms, attr_name) == exists
 
 
+# TODO/RepoSplit: This was changed to only run against the mock package repo
 @pytest.mark.regression("24552")
-def test_all_package_names_is_cached_correctly():
+def test_all_package_names_is_cached_correctly(mock_packages):
     assert "mpi" in spack.repo.all_package_names(include_virtuals=True)
     assert "mpi" not in spack.repo.all_package_names(include_virtuals=False)
 
 
+# TODO/RepoSplit: Created a separate mock repo with the desired package.
+@pytest.fixture()
```

Comment:
I don't see where this fixture is being used, I can't tell whether it's vestigial or I'm missing something.

Reply:
Thanks for catching that. It was residual from an investigation.